### PR TITLE
Python 3.11 + Hatch + Latest version for all dependencies + Fix tests

### PR DESCRIPTION
**Pull request by ouziel-slama**
- [x] Replace `setup.py` with `pyproject.toml`.
- [x] `counterparty-lib` requires now Python >= 3.11.
- [x] Using the latest versions for all dependencies.
- [x] Update `circle.yml` and `README.md`
- [x] Fix regular tests.
- [x] Replace `--skiptestbook` by `--testbook`
- [x] Add `--testbookblockindex` to test book only from the given block 
- [x] Temporarily deactivates the test book
- [x] Fix https://github.com/CounterpartyXCP/counterparty-lib/issues/1134

To install the library just use `pip install .` or `pip install -e .` in editable mode.

You can also use hatch to run test in a venv.
```
$ hatch run pytest counterpartylib/test/ --skiptestbook=all
```

### CODE REVIEW

**Comment by robby-dermody** [comment]
nice! welcome back, @ouziel-slama !

**Comment by fractastical** [comment]
Welcome @ouziel!

File: counterpartylib/test/fixtures/scenarios.py [hunk 1]
```diff
@@ -29,13 +29,14 @@
...
 
+
 UNITTEST_FIXTURE = [
     ['burn', (ADDR[0], DP['burn_quantity']), {'encoding': 'multisig'}],  # 310000
-    ['issuance', (ADDR[0], None, 'DIVISIBLE', DP['quantity'] * 1000, True, 'Divisible asset'), {'encoding': 'multisig'}],
-    ['issuance', (ADDR[0], None, 'NODIVISIBLE', 1000, False, 'No divisible asset'), {'encoding': 'multisig'}],
-    ['issuance', (ADDR[0], None, 'CALLABLE', 1000, True, 'Callable asset'), {'encoding': 'multisig'}],
-    ['issuance', (ADDR[0], None, 'LOCKED', 1000, True, 'Locked asset'), {'encoding': 'multisig'}],
-    ['issuance', (ADDR[0], None, 'LOCKED', 0, True, 'LOCK'), {'encoding': 'multisig'}],
+    ['issuance', (ADDR[0], None, 'DIVISIBLE', DP['quantity'] * 1000, True, 'Divisible asset', None, None), {'encoding': 'multisig'}],
```
Comment (adamkrellenstein) [inline counterpartylib/test/fixtures/scenarios.py:35]: Do we need this if those two additional issuance params are optional?

**Comment by adamkrellenstein** [comment]
This is looking beautiful!!!

File: counterpartylib/test/fixtures/scenarios.py (hunk 1 above)
Comment (ouziel-slama) [reply counterpartylib/test/fixtures/scenarios.py:35]: yep! it's an unsuccessful try to fix a test :) I think there is missing parameters but not them!

**Comment by jdogresorg** [comment]
amazing work @ouziel-slama !

**Comment by ouziel-slama** [comment]
Ready to merge for me. I added the Monaparty fix for https://github.com/CounterpartyXCP/counterparty-lib/issues/1134 with some testing.

**Pull request merged**